### PR TITLE
support configuring deny.txt

### PR DESCRIPTION
**Pull request by etam**

### CODE REVIEW

File: templates/deny.txt.j2 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{% for node_id in golem_source_deny_nodes %}
```
Comment (ktomala) [inline templates/deny.txt.j2:1]: Consider using {%- ... %} so there won't be empty newline

File: templates/deny.txt.j2 [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+{{ node_id }}
+{% endfor %}
```
Comment (ktomala) [inline templates/deny.txt.j2:3]: Consider using {%- ... %} so there won't be empty newline

**Review by ktomala — CHANGES_REQUESTED** [review]

File: templates/deny.txt.j2 (hunk 1 above)
Comment (etam) [reply templates/deny.txt.j2:1]: What empty newline? This creates a nice file with every node_id in a separate line. AFAIK your solution would join them all in a single line.

Comment (ktomala) [reply templates/deny.txt.j2:1]: Jinja2 will put newlines in lines where {% ... %} is.

Comment (ktomala) [reply templates/deny.txt.j2:1]: Take an example:
In [1]: import jinja2

In [2]: tpl = """{% for node_id in golem_source_deny_nodes %}
   ...: {{ node_id }}
   ...: {% endfor %}
   ...: """

In [3]: golem_source_deny_nodes = ['one', 'two', 'three', 'four']

In [4]: t = jinja2.Template(tpl)

In [5]: t.render(golem_source_deny_nodes=golem_source_deny_nodes)
Out[5]: '\none\n\ntwo\n\nthree\n\nfour\n'

In [6]: print(t.render(golem_source_deny_nodes=golem_source_deny_nodes))

one

two

three

four

Comment (ktomala) [reply templates/deny.txt.j2:1]: And this is how it behaves with {%-

In [24]: tpl = """{%- for node_id in golem_source_deny_nodes %}
    ...: {{ node_id }}
    ...: {%- endfor %}
    ...: """

In [25]: t = jinja2.Template(tpl)

In [26]: t.render(golem_source_deny_nodes=golem_source_deny_nodes)
Out[26]: '\none\ntwo\nthree\nfour'

In [27]: print(t.render(golem_source_deny_nodes=golem_source_deny_nodes))

one
two
three
four

Comment (etam) [reply templates/deny.txt.j2:1]: Ansible uses by default
```
t = jinja2.Template(tpl, trim_blocks=True)
```

Comment (etam) [reply templates/deny.txt.j2:1]: The docs says it's set to 'no', but that's not true.
https://github.com/ansible/ansible/pull/40312

Comment (ktomala) [reply templates/deny.txt.j2:1]: ok, I will approve this then

**Review by ktomala — APPROVED** [review]

**Pull request merged**